src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -105,7 +105,7 @@ def process_request(self, request):
                                               quote_plus(self.moniker))
 
         if is_text_payload(request) and request.body:
-            body = str(request.body)
+            body = str(request.body, 'utf-8') if isinstance(request.body, bytes) else str(request.body)
```

Comment:
[bytes is an alias for str](https://docs.python.org/3/whatsnew/2.6.html#pep-3112-byte-literals) from python 2.6 onwards in python 2, so this won't NameError